LightGBM/data.py

```diff
@@ -124,6 +169,40 @@ def features(df):
         else:
             df[name] = getattr(df["date"].dt, attribute).astype("int16")
 
+    df.dropna(inplace = True)
+    return df
+
+
+def training_data(df):  
+    drop_cols = ["id", "date", "sales", "d", "wm_yr_wk", "weekday"]
+    keep_cols = df.columns[~df.columns.isin(drop_cols)]
+
+    X = df[keep_cols]
+    y = df["sales"]
+
+    return X, y
+
+
+def lgb_dataset(calendar, prices, sales):
```

Comment:
no docs

Reply:
This was an old commit, I added it in a later commit.